### PR TITLE
Fix Equals of AutoCompleteResponse and simplify some code

### PR DESCRIPTION
**Pull request by ShuiRuTian**

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/OmniSharp/omnisharp-roslyn?pullRequest=2362) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/OmniSharp/omnisharp-roslyn?pullRequest=2362) before we can accept your contribution.<br/><br/>:x: ShuiRuTian [sign now](https://cla.dotnetfoundation.org/OmniSharp/omnisharp-roslyn?pullRequest=2362)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/OmniSharp/omnisharp-roslyn?pullRequest=2362) it.</sub>

File: src/OmniSharp.Abstractions/Models/v1/AutoComplete/AutoCompleteResponse.cs [hunk 1]
```diff
@@ -24,7 +24,8 @@ public class AutoCompleteResponse
         public override bool Equals(object other)
         {
             var otherResponse = other as AutoCompleteResponse;
-            return otherResponse.DisplayText == DisplayText
+            return otherResponse != null
+                && otherResponse.DisplayText == DisplayText
```
Comment (333fred) [inline src/OmniSharp.Abstractions/Models/v1/AutoComplete/AutoCompleteResponse.cs:28]: I believe this is the only actual change in this PR. Please revert all the other changes: I'd be fine with them in a separate PR, but the large number of unrelated changes makes it more difficult to understand what the bug is and what changed here. I'd also like to see a test showing what bugs this was causing.

**Review by 333fred — COMMENTED** [review]
Who is actually still using the AutoComplete endpoint, and can they move to the completion endpoint?

**Comment by nickspoons** [comment]
> Who is actually still using the AutoComplete endpoint, and can they move to the completion endpoint?

OmniSharp-vim uses it still. We can look at moving to the new completion endpoint but it won't be today :smile:

File: src/OmniSharp.Abstractions/Models/v1/AutoComplete/AutoCompleteResponse.cs (hunk 1 above)
Comment (ShuiRuTian) [reply src/OmniSharp.Abstractions/Models/v1/AutoComplete/AutoCompleteResponse.cs:28]: I am not sure whether this deserve a test. I did not meet any real world case and just did the changes by the way when reading code.
Anyway, this shows the difference:
```C#
    public unsafe static void Main()
    {
        AutoCompleteResponse foo1 = new();
        Object foo2 = new();
        TestEqual(foo1, foo2); // before the change, console logs "error", for we try to access a property of null.
                               // after the change, console logs "not equal"
    }

    public static void TestEqual(Object o1, Object o2)
    {
        try
        {
            if (!o1.Equals(o2))
            {
                Console.WriteLine("not equal");
            }
        }
        catch
        {
            Console.WriteLine("error");
        }
    }
```

Comment (333fred) [reply src/OmniSharp.Abstractions/Models/v1/AutoComplete/AutoCompleteResponse.cs:28]: Alright. Please revert the rest of the changes though.

**Review by 333fred — APPROVED** [review]
